### PR TITLE
refactor/test/fix: liquidity pricer

### PR DESCRIPTION
**Pull request by p0mvn**
Refactor & progress towards pool liquidity capitalization/pricing

The core of the change focuses on creating better `PriceCoin` and `PriceBalances` APIs on the liquidity price and covering them with more tests.

Extracted from: https://github.com/osmosis-labs/sqs/pull/332 for ease of review.

As an outcome, identified and fixed an overapplying quote denom scaling factor bug in edge cases. It was unnoticed since this logic is not relied on in production yet.

Tested on the node

### CODE REVIEW

File: domain/pricing.go [hunk 1]
```diff
@@ -175,22 +170,27 @@ type PoolLiquidityPricerWorker interface {
 	RegisterListener(listener PoolLiquidityComputeListener)
 }
 
-// DenomPriceInfo defines the price information for the base denom.
 type DenomPriceInfo struct {
-	// Price is the price of the base denom.
-	Price osmomath.BigDec
-	// ScalingFactor is the scaling factor for the base denom.
+	Price         osmomath.BigDec
```
Comment (cryptomatictrader) [inline domain/pricing.go:174]: Just curious - what is the general practice when using osmomath vs native (coz I think I will face some issue when doing integration test for dexscreener when we have data passing around node => pubsub => firestore) ? For price, can we not use osmomath.BigDec and just native data type?

Comment (cryptomatictrader) [reply domain/pricing.go:174]: I see in most of our code, we tend to use osmomath by default vs native data type, and I think there must be some reasons for that (added convenient methods?)

**Review by cryptomatictrader — APPROVED** [review]
LGTM. Thank you!

File: domain/pricing.go (hunk 1 above)
Comment (p0mvn) [reply domain/pricing.go:174]: There are 2 reasons:
- We need cross-platform determinism that native floats do not provide. As a result, we use fixed point precision decimal types
- Size of the data. Prices can be large enough to exceed the max float value

Here in SQS, since we depend on the chain APIs, we simply inherit all of the types for simplicity.
